Moshe

MosheBrizel opened the pull request:
add the option to borrow a book.
add option to delete the database and to create.

MosheBrizel committed:
test

BenZionHasida committed:
test to learn git

BenZionHasida committed:
test to learn git

BenZionHasida committed:
test to learn git

BenZionHasida committed:
test to learn git

BenZionHasida committed:
fix some spelling mistakes

BenZionHasida committed:
add a comments

MosheBrizel committed:
change the database
1.remove the number of the books and add a column of accompanied.
2. add a please to chack the database.

MosheBrizel committed:
change the database
1.remove the number of the books and add a column of accompanied.
2. add a please to chack the database.

MosheBrizel committed:
add the option to take a book with search the book

NoamAnisfeld committed:
Revert "add a comments"

This reverts commit d5bd6c4047831b2bb14598328f06f2c00520c3ad.

NoamAnisfeld committed:
Revert "fix some spelling mistakes"

This reverts commit 684c2e84e5ecb371a3054fc151c3ca56338eefbb.

NoamAnisfeld committed:
Revert "test to learn git"

This reverts commit c4de94abe9883c4e29bbf375a88a2747b774aefb.

NoamAnisfeld committed:
Revert "test to learn git"

This reverts commit 54ee03b08782d51d52deec9e82de2dd148118b5d.

NoamAnisfeld committed:
Revert "test to learn git"

This reverts commit 3726f7b38bfdf4d5718bdb22adda659360ebc08b.

NoamAnisfeld committed:
Revert "test to learn git"

This reverts commit 6696f62bdd6a9f6330695ea7d74255dd0511b334.

NoamAnisfeld committed:
Revert "test"

This reverts commit 6b40c38100561f9f09f29f12c60b7a46122efa8c.

NoamAnisfeld committed:
Revert "test"

This reverts commit 6b40c38100561f9f09f29f12c60b7a46122efa8c.

MosheBrizel committed:
changes
1. an option to delete the database and to create the database.
2. add an option to change a column in a row in the database.

MosheBrizel committed:
sample changes

MosheBrizel committed:
Merge remote-tracking branch 'origin/Moshe' into Moshe

MosheBrizel committed:
change to borrow a book,
1. add a html page.
2. add a path in the flask page.
3. add a js file to control the all books on the clint page, and to send a post request to the server.

MosheBrizel committed:
fixes a mistake of the button to borrow a book.

MosheBrizel committed:
fixes a mistake of the flask what the git changed.

NoamAnisfeld commented:
Thanks!

### File `take_book.js`
1.
```js
listBooks.forEach((book) => {
  if (book.getAttribute("borrowed") == "True") {
    // To show.
    book.style.display = "list-item";
  } else {
    // To not show
    book.style.display = "none";
  }
});
```

This can be achieved by CSS instead:
```css
.book:not([borrowed="True"] {
  display: none;
}
```

2.
```js
inputBook.addEventListener("input", () => {
  const text = inputBook.value.toLowerCase();
  listBooks.forEach((book) => {
    const bookTitle = book.textContent.toLowerCase();
    if (bookTitle.includes(text) && book.getAttribute("borrowed") == "True") {
      // To show.
      book.style.display = "list-item";
    } else {
      // To not show.
      book.style.display = "none";
    }
  });
});
```


Perhaps it can be a bit simpler
```js
listBooks = document.querySelectorAll('.book:not([borrowed="True"])';
inputBook.addEventListener("input", () => {
  const text = inputBook.value.toLowerCase();
  listBooks.forEach((book) => {
    const bookTitle = book.textContent.toLowerCase();
    if (bookTitle.includes(text)) {
      book.hidden = true;
    } else {
      book.hidden = false;
    }
  });
});
```
(`hidden` in HTML is pretty much like `display: none;` in CSS. Alternatively you could also use a "hidden" class and hide with CSS or toggle between `book.style.display = "none"` and `book.style.display = null' to revert it)


3.
```js
function sendDateToTheServer(data, action) {
  const xhr = new XMLHttpRequest();
  xhr.open("POST", action, true);
  xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
  // Send the POST request with the data
  xhr.send("data=" + encodeURIComponent(data));
  xhr.onreadystatechange = function () {
    if (xhr.readyState === 4 && xhr.status === 200) {
      // Handle the response from the server
      const response = JSON.parse(xhr.responseText);
      console.log(response);
      if (response["data"] == "true") {
        window.location.href = "/show_books";
      }
    }
  };
}
```
 This is an old method in JS, you should use `fetch` instead:
```js
function sendDateToTheServer(data, action) {
  fetch(action, {
    method: "POST",
    headers: {
      "Content-Type", "application/x-www-form-urlencoded"
    },
    body: "data=" + encodeURIComponent(data),
  })
  .then(res => res.json())
  .then(res => {
    if (res.data === "true") {
      window.location.href = "/show_books";
    }
  });
}
```

### Other
Why `borrowed="True"` for books that are not borrowed yet? shouldn't it be the opposite?

NoamAnisfeld commented:
Thanks!

### File `take_book.js`
1.
```js
listBooks.forEach((book) => {
  if (book.getAttribute("borrowed") == "True") {
    // To show.
    book.style.display = "list-item";
  } else {
    // To not show
    book.style.display = "none";
  }
});
```

This can be achieved by CSS instead:
```css
.book:not([borrowed="True"] {
  display: none;
}
```

2.
```js
inputBook.addEventListener("input", () => {
  const text = inputBook.value.toLowerCase();
  listBooks.forEach((book) => {
    const bookTitle = book.textContent.toLowerCase();
    if (bookTitle.includes(text) && book.getAttribute("borrowed") == "True") {
      // To show.
      book.style.display = "list-item";
    } else {
      // To not show.
      book.style.display = "none";
    }
  });
});
```

Perhaps it can be a bit simpler
```js
listBooks = document.querySelectorAll('.book[borrowed="True"])';
inputBook.addEventListener("input", () => {
  const text = inputBook.value.toLowerCase();
  listBooks.forEach((book) => {
    const bookTitle = book.textContent.toLowerCase();
    if (bookTitle.includes(text)) {
      book.hidden = true;
    } else {
      book.hidden = false;
    }
  });
});
```
(`hidden` in HTML is pretty much like `display: none;` in CSS. Alternatively you could also use a "hidden" class and hide with CSS or toggle between `book.style.display = "none"` and `book.style.display = null` to revert it)

3.
```js
function sendDateToTheServer(data, action) {
  const xhr = new XMLHttpRequest();
  xhr.open("POST", action, true);
  xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
  // Send the POST request with the data
  xhr.send("data=" + encodeURIComponent(data));
  xhr.onreadystatechange = function () {
    if (xhr.readyState === 4 && xhr.status === 200) {
      // Handle the response from the server
      const response = JSON.parse(xhr.responseText);
      console.log(response);
      if (response["data"] == "true") {
        window.location.href = "/show_books";
      }
    }
  };
}
```

this is an old method in JS, you should be using `fetch` instead:
```js
function sendDateToTheServer(data, action) {
  fetch(action, {
    method: "POST",
    headers: {
      "Content-Type", "application/x-www-form-urlencoded"
    },
    body: "data=" + encodeURIComponent(data),
  })
  .then(res => res.json())
  .then(res => {
    if (res.data === "true") {
      window.location.href = "/show_books";
    }
  });
}
```

### Other
- JSON has boolean values `true` and `false`, better to use them than a `"true"` string
- `borrowed="True"` seems a bit strange for marking books that are *not* borrowed yet
- Non-standard HTML attributes should be prefixed with `data-`, for example `data-borrowed="True"`. It works without it but is considered invalid HTML and might cause bugs in some cases

MosheBrizel committed:
1 I change the 3 changes
2. in the other
 -I fixe "true" to True
 -"borrowed" change to False
 - add "data-" to borrowed
